systemtest/src/test/java/io/strimzi/systemtest/performance/TopicOperatorPerformance.java

```diff
@@ -740,6 +743,136 @@ void testCapacity(String maxBatchSize, String maxBatchLingerMs) throws IOExcepti
+                                    .addNewEnv()
+                                        .withName("STRIMZI_MAX_BATCH_SIZE")
+                                        .withValue(maxBatchSize)
+                                    .endEnv()
+                                    .addNewEnv()
+                                        .withName("MAX_BATCH_LINGER_MS")
+                                        .withValue(maxBatchLingerMs)
+                                    .endEnv()
+                                .endTopicOperatorContainer()
+                            .endTemplate()
+                        .endEntityOperator()
+                    .endSpec()
+                    .build(),
+                ScraperTemplates.scraperPod(testStorage.getNamespaceName(), testStorage.getScraperName()).build()
+            );
+
+            this.testStorage.addToTestStorage(TestConstants.SCRAPER_POD_KEY,
+                kubeClient().listPodsByPrefixInName(this.testStorage.getNamespaceName(), testStorage.getScraperName()).get(0).getMetadata().getName());
+
+            // -- Metrics POLL --
+            // Assuming 'testStorage' contains necessary details like namespace and scraperPodName
+            this.topicOperatorCollector = new TopicOperatorMetricsCollector.Builder()
+                .withScraperPodName(this.testStorage.getScraperPodName())
+                .withNamespaceName(this.testStorage.getNamespaceName())
+                .withComponentType(ComponentType.TopicOperator)
+                .withComponentName(this.testStorage.getClusterName())
+                .build();
+
+            this.topicOperatorMetricsGatherer = new TopicOperatorMetricsCollectionScheduler(this.topicOperatorCollector, "strimzi.io/cluster=" + this.testStorage.getClusterName());
+            this.topicOperatorMetricsGatherer.startCollecting();
+
+            long startTime = System.currentTimeMillis();
+
+            TopicOperatorPerformanceUtils.processKafkaTopicBatchesAsync(testStorage, numberOfTopics);
+
+            // Calculate total execution time
+            allTasksTimeMs = System.currentTimeMillis() - startTime;
+
+            LOGGER.info("Total time taken for all tasks (i.e., {} for each operation; creation, modification and deletion) {} ms", numberOfTopics, allTasksTimeMs);
+        } finally {
+            // to enchantment a process of deleting we should delete all resources at once
+            // I saw a behaviour where deleting one by one might lead to 10s delay for deleting each KafkaTopic
+            LOGGER.info("Start deletion KafkaTopics in namespace:{}", testStorage.getNamespaceName());
+            resourceManager.deleteResourcesOfTypeWithoutWait(KafkaTopic.RESOURCE_KIND);
+            KafkaTopicUtils.waitForTopicWithPrefixDeletion(testStorage.getNamespaceName(), testStorage.getTopicName());
+            if (this.topicOperatorMetricsGatherer != null) {
+                this.topicOperatorMetricsGatherer.stopCollecting();
```

Comment:
We should stop collecting metrics just after `processKafkaTopicBatchesAsync`. Deleting topics in the `finally` block is just a safe net in case something goes wrong and some KT are not deleted.

Reply:
> Deleting topics in the finally block is just a safe net in case something goes wrong and some KT is not deleted.

Yeah, true I will re-order this two ;)